Generic_CCVA.Rmd

```diff
@@ -126,65 +126,73 @@ watersupply_flowlines <- watersupply_watershed %>%
 
 ```
 
-#### Centroids
+##### Centroids
 
-Climate futures have been previously compiled for park and Koppen centroids. Check which f these best represents conditions at the (potential) source(s) for the park's water supply.
+Climate futures have been previously compiled for park and Koppen centroids. Check which of these best represents conditions at the (potential) source(s) for the park's water supply.
+
+https://www.gloh2o.org/koppen/
+
+Beck HE, Zimmermann NE, McVicar TR, Vergopolan N, Berg A, Wood EF. Present and future Köppen-Geiger climate classification maps at 1-km resolution. Sci Data. 2018 Oct 30;5:180214. doi: 10.1038/sdata.2018.214. Erratum in: Sci Data. 2020 Aug 17;7(1):274. PMID: 30375988; PMCID: PMC6207062.
+
+#### First get Koppen climate data for park
 
 ```{r, eval = TRUE, echo = FALSE, message = FALSE, warning = FALSE}
 
 # Get koppen climate grid for park with buffer
-koppen_park <- get_koppen(park = park, 
-                             aoi = park_boundary, 
+koppen_park <- get_koppen(aoi = park_boundary, 
                              buffer_dist = 0.1)
 
 
 # Get koppen climate intersecting watershed
 # create a merged watersupply watershed for subsequent functions
-ws_union <- st_union(watersupply_watershed) %>%
-            st_as_sf()
+ws_union <- sf::st_union(watersupply_watershed) %>%
+            sf::st_as_sf()
 
-koppen_ws <- get_koppen(park = park, 
-                           aoi = ws_union, 
+koppen_ws <- get_koppen(aoi = ws_union, 
                            buffer_dist = 0) %>%
             sf::st_transform(., crs = sf::st_crs(ws_union)) %>%
             dplyr::filter(st_intersects(., ws_union) %>% 
             as.logical())
             
 
-# identify the koppen climate that spans the greatest number of cells
-# in watershed area
-koppen_ws_sel <- koppen_ws %>%
-              dplyr::count(K_vals)
-
 # Create a table of Koppen and park centroids with area span for each koppen
 # climate that overlaps with watersupply watershed (from above) and distance
 # between centroid location and watersupply watershed. 
 
 #****KEC:  Note, nothing is done with this as of now, but we can incorporate 
 #* later
-url <- 'https://parkfutures.s3.us-west-2.amazonaws.com/maca-tprh-data/0Climate_Zones_by_Unit_cleaned.csv'
-centroid_sel <- GET(url) %>% 
+
+centroid_sel <- 
+  # get table of koppen centroids from web
+  GET('https://parkfutures.s3.us-west-2.amazonaws.com/maca-tprh-data/0Climate_Zones_by_Unit_cleaned.csv') %>% 
             as.character() %>% 
             data.table::fread(.,header = TRUE, skip = 1) %>%
             dplyr::filter(UNIT_CODE == park) %>%
             sf::st_as_sf(coords = c("Lon","Lat"), 
                         crs = sf::st_crs(watersupply_watershed)) %>%
```

Comment:
Will want to double check what the CRS for the centroids are - are they in fact the same as the delineated WSs?